internal/generator/fluentd/output/fluentdforward/fluentdforward.go

```diff
@@ -94,7 +94,7 @@ func SecurityConfig(o logging.OutputSpec, secret *corev1.Secret) []generator.Ele
 		conf = []generator.Element{
 			TLS(true),
 		}
-		if secret == nil {
+		if o.TLS != nil && o.TLS.InsecureSkipVerify {
 			conf = append(conf, TLS(false))
 		}
 	}
```

Comment:
SkipVerify does not mean turn off TLS, it means add the required configuration to enable TLS but skip cert verify. Also the old code looks wrong anyway - TLS on/off depends on the schema (http vs. https) not on the presence or absence of a secret. I think it should be:
```
	if url.IsTLSScheme(u.Schema) {
		conf = append(conf, TLS(true))
		conf = append(conf, TLSSkipVerify(o.TLS != nil && o.TLS.InsecureSkipVerify))
	} else {
		conf = append(conf, TLS(false))
	}
```

Reply:
The version you commented on is outdated by now. `TLS(false)` was a bit unclear in what it was actually doing. I assumed that it turns off TLS at first as well (and it does for other output implementations). In this case the template for `TLS(false)` just generated the `insecure_mode true` flag, that's why both `TLS(true)` and `TLS(false)` were added to the configuration.

I have refactored the `TLS` type to be less ambiguous in the current state of the PR.